src/quickgui/plugin/qgsquickfeatureform.qml

```diff
@@ -0,0 +1,470 @@
+/***************************************************************************
+ qgsquickfeatureform.qml
+  --------------------------------------
+  Date                 : Nov 2017
+  Copyright            : (C) 2017 by Matthias Kuhn
+  Email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+import QtQuick 2.6
+import QtQuick.Controls 2.0
+import QtQuick.Dialogs 1.2
+import QtQuick.Layouts 1.3
+import QtGraphicalEffects 1.0
+import QtQml.Models 2.2
+import QtQml 2.2
+
+// We use calendar in datetime widget that is not
+// yet implemented in Controls 2.2
+import QtQuick.Controls 1.4 as Controls1
+
+import QgisQuick 0.1 as QgsQuick
+
+Item {
+  signal saved
+  signal canceled
+  signal aboutToSave
+
+  property QgsQuick.AttributeFormModel model
+  property alias toolbarVisible: toolbar.visible
+  property bool allowRememberAttribute: false // when adding new feature, add checkbox to be able to save the same value for the next feature as default
```

Comment:
Is there a plan to do something doxygen-alike for qml documentation? Would be great if we had a stable format for api docs right from the beginning.

Reply:
Would be great, but it would take a bit of investigation on how to do it properly, what tool to use to generate the docs etc. I would rather keep it for future enhancement

Reply:
Ok. I did a quick search and I think Qt does it via .qdoc files (not within the qml files). Let's keep eyes open and postpone this.